mockwebserver/src/main/java/okhttp3/mockwebserver/MockWebServer.kt

```diff
@@ -329,8 +329,7 @@ class MockWebServer : ExternalResource(), Closeable {
    * @return the head of the request queue
    */
   @Throws(InterruptedException::class)
-  fun takeRequest(timeout: Long, unit: TimeUnit): RecordedRequest? =
-      requestQueue.poll(timeout, unit)
+  fun takeRequest(timeout: Long, unit: TimeUnit): RecordedRequest = requestQueue.poll(timeout, unit)
```

Comment:
Can we change the return type to non-nul here? 🤔

Reply:
Yes we can!

Reply:
This changes it to a platform type, not non-null. We should switch it back.

Reply:
I changed it to explicitly return `RecordedResponse`, which is non-null. Definitely need to avoid mystery types.

Reply:
There's probably a lint or something we can enable. Android's lint has a check and works on Java projects. I'll look into it later.

Reply:
Doesn't seem to run on Kotlin sources. Filed https://issuetracker.google.com/issues/134237547